src/frontend/discussions-abusefilter-app/src/interfaces/BackendInterface.ts

```diff
@@ -0,0 +1,9 @@
+import type { Filter, FilterMetadata } from '@shared/filters';
+
+export interface BackendInterface {
+  getFilters: <T = any>() => Promise<T> | FilterMetadata[],
+  getFilter: <T = any>(filterId: string) => Promise<T> | { filterDetails: Filter | undefined, filterMetadata: FilterMetadata | undefined },
```

Comment:
These should never be undefined - if undefined, something's gone very wrong.

I think it's common at Fandom to return an error message, so it can be either all defined `| Error` e.g. 

```ts
type Error = {
  error: {
    message: string
  }
}
```